source/extensions/stat_sinks/hystrix/hystrix.h

```diff
@@ -0,0 +1,144 @@
+#pragma once
+
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "envoy/server/admin.h"
+#include "envoy/server/instance.h"
+#include "envoy/stats/stats.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace StatSinks {
+namespace Hystrix {
+
+typedef std::vector<uint64_t> RollingWindow;
+typedef std::map<const std::string, RollingWindow> RollingStatsMap;
+
+struct ClusterStatsCache {
+  // This constructor initializes all the stat names by concatenating
+  // the cluster name and the rest of the stat name that's required for
+  // the lookup.
+  ClusterStatsCache(const std::string& cluster_name);
+
+  std::string errors_name_;
```

Comment:
Are these names just here for the purpose of testing and logging?  I don' think the full stat names are still needed since we're looking the stats up within the cluster scope.  It appears that they're only used in `printRollingWindows()`, which seems to only be used in tests and log messages?  Not sure how others feel, but I think we should probably avoid storing this state if it's just for logging and testing.  If we want to keep that method for logging and testing, we could just let it construct whatever preferred stat name is needed on the fly.

Reply:
Pushed new version. Please see if it seems more reasonable now. Another option is to totally remove it. It is there just for logging. Not needed for testing.